protons/app/driver.py

```diff
@@ -777,6 +793,22 @@ def __init__(self, temperature, topology, system, pressure=None, perturbations_p
         self.ncmc_integrator = None
         self.context = None
 
+        # A salt swap swapper can later be attached to enable counterion coupling to protonation state changes
+        # Using the `attach_swapper` method
+        self.swapper = None
+        # The total excess charge from ions, applied as counter-charge to protonation state changes.
+        # Positive indicates the amount of cations that have been added to the system
+        # Negative indicates the amount of anions that have been added to the system
+        # The drive should never add cations and anions at the same time.
+        self.excess_ions = 0
```

Comment:
Do you want `excess_ions` to account for the neutralizing counterions that are added by `salinator` before the simulation? For example, when `swapper` is first attached to `driver`, you could count the excess charge of the counterions.

Reply:
I changed to code to use the count from the swapper, thanks for the suggestion!